src/main/java/use_case/getpost/GetPostDataAccessInterface.java

```diff
@@ -15,12 +15,18 @@ public interface GetPostDataAccessInterface {
      * @return the Post object with the given entryID
      * @throws PostNotFoundException if a post with the given entryID does not exist
      */
-    Post getPostByEntryID(String entryID) throws PostNotFoundException;
+    JSONObject getPostByEntryID(String entryID) throws PostNotFoundException;
 
     /**
      * Retrieve all posts made by a given user  post with a specific entryID.
      * @param userID the userID of the user to retreive posts
-     * @return a list of Post objects by the given user
+     * @return a list of Post json data of the given user
      */
-    List<Post> getAllPostsByUserID(String userID);
+    List<JSONObject> getAllPostsByUserID(String userID);
+
+    /**
+     * Retrieve all posts in the database.
+     * @return a list of Post objects
+     */
+    List<JSONObject> getAllPosts();
```

Comment:
The comment says a list of Post objects, but the generic type is JSONObject. Does this need to be changed?
```suggestion
    List<Post> getAllPosts();
```

Reply:
yup! i forgot to update that part of the docs!